numpy/ma/core.py

```diff
@@ -5134,34 +5134,36 @@ def var(self, axis=None, dtype=None, out=None, ddof=0,
 
         # Easy case: nomask, business as usual
         if self._mask is nomask:
-            return self._data.var(axis=axis, dtype=dtype, out=out,
-                                  ddof=ddof, **kwargs)
-        # Some data are masked, yay!
-        cnt = self.count(axis=axis, **kwargs) - ddof
-        danom = self - self.mean(axis, dtype, keepdims=True)
-        if iscomplexobj(self):
-            danom = umath.absolute(danom) ** 2
+            dvar = super(MaskedArray, self).var(axis=axis, dtype=dtype,
+                                                ddof=ddof, **kwargs)[()]
         else:
-            danom *= danom
-        dvar = divide(danom.sum(axis, **kwargs), cnt).view(type(self))
-        # Apply the mask if it's not a scalar
-        if dvar.ndim:
-            dvar._mask = mask_or(self._mask.all(axis, **kwargs), (cnt <= 0))
-            dvar._update_from(self)
-        elif getattr(dvar, '_mask', False):
-            # Make sure that masked is returned when the scalar is masked.
-            dvar = masked
-            if out is not None:
-                if isinstance(out, MaskedArray):
-                    out.flat = 0
-                    out.__setmask__(True)
-                elif out.dtype.kind in 'biu':
-                    errmsg = "Masked data information would be lost in one or "\
-                             "more location."
-                    raise MaskError(errmsg)
-                else:
-                    out.flat = np.nan
-                return out
+            # Some data are masked, yay!
+            cnt = self.count(axis=axis, **kwargs) - ddof
+            danom = self - self.mean(axis, dtype, keepdims=True)
+            if iscomplexobj(self):
+                danom = umath.absolute(danom) ** 2
```

Comment:
Good to check for a complex object, but might as well do (there was a PR to make a ufunc out of this...):
```
danom = danom.real**2 + danom.imag**2
```